src/LSST/control/PowerSubsystem.cpp

```diff
@@ -463,6 +450,13 @@ void PowerSubsystem::_setPowerOff(string const& note) {
     }
 }
 
+void PowerSubsystem::_reportStateChange() const {
+    auto context = _context.lock();
```

Comment:
This is not safe since you're setting `_context` externally via `PowerSubsystem::setContext(...)`. Perhaps there is a possibility to redesign the class (and others) to pass the context to the c-tor? Another (potential_ problem with such `set`-ter methods is thread (un-)safety of this patter as you would have to make sure things are properly synchronized before/while used.

Reply:
Once set, context never changes, at least until the number of shared pointers drops to zero. Once the reference count has dropped to zero, the weak pointer lock() will fail. 

It's very important that `Context::_thisPtr`, and all the other static `_thisPtr`s never change once set. This means the pointers can be treated as constants, which makes dealing with them much easier.

Reply:
Also, for simplicity, some unit tests don't ever define Context, and without that there's nowhere to report the change.